go/private/tools/overlay_repository.bzl

```diff
@@ -93,7 +93,7 @@ def _apply_overlay(ctx, overlay):
 
     # TODO(jayconrod): sanitize destination paths.
     for src_path, dst_rel in overlay:
-        _check_execute(ctx, ["cp", src_path, dst_rel], "failed to copy file from %s" % src_path)
+        ctx.symlink(src_path, dst_rel)
```

Comment:
I'm not confident a symlink will always work here: I've been told they may require administrative privilege to create on NTFS. Also, we may need to replace files, and I don't think `ctx.symlink` does that. I don't have my Windows machine with me at the moment, so I can't test it out right now.

I was tinkering with this a few days ago, and the best thing I came up with was using the `BAZEL_SH` environment variable to locate msys2 bash, then looking in the same directory for cp. However, it seems like the requirement to set `BAZEL_SH` has been removed from the installation instructions, so maybe I shouldn't rely on that.

Maybe the best thing to do is create a .bat file with `copy` commands, then execute that?

In any case, this file was copied from `github.com/bazelbuild/bazel-gazelle/internal/overlay_repository.bzl`. It should be modified there first, and the changes should be backported here.

Reply:
Yes. Specifically, SeCreateSymbolicLinkPrivilege is a right given by default only to users in the Administrators group.

Reply:
`rules_go` uses `repository_ctx.symlink` already in a few places. Bazel uses it all over the place. I believe bazel actually creates junctions for directories and copies for files. Either way, without  `repository_ctx.symlink` nothing would work.

Reply:
I've sent https://github.com/bazelbuild/bazel-gazelle/pull/246, making the same change to bazel-gazelle.